ui/eureka/eureka.js

```diff
@@ -2139,6 +2611,48 @@ class EurekaTracker {
     }
   }
 
+  OnCE(e) {
+    // Upon entering Eureka we usually receive the CE info before
+    // this.zoneInfo is loaded, so lets store the events until we're
+    // able to process them.
+
+    if (!this.zoneInfo) {
+      this.CEQueue.push(e);
+      return;
+    }
+
+    switch (e.detail.eventType) {
```

Comment:
This looks quite repetitive, could something like this work?
```js
const handleEvent = (handler) => {
  for (const key of this.nmKeys) {
    const nm = this.nms[key];
    if (e.detail.data.ceKey == nm.ceKey) {
      return handler(nm);
    }
  }
};

switch (e.detail.eventType) {
  case "add":
    handleEvent(this.OnFatePop);
    break;
  case "remove":
    handleEvent(this.OnFateKill);
    break;
  case "update":
    handleEvent(
      (nm) =>
        e.detail.data.status === 3 &&
        this.OnFateUpdate(nm, e.detail.data.progress)
    );
}
```

Reply:
This looks pretty neat! Thanks for the suggestion, Sorry for taking a while to reply.